### DISCUSSION TITLE
Display Error UTF-8 to ASCII

### DISCUSSION THREAD

**Discussion by Archie1978** [sourcelair/xterm.js#4481] [Q&A] [answered]
Hello,
   I use the version 5.1.0 and I try to write into terminal. I have the "é".I have the character that is displayed in 2 characters but I can't find  where is my error.
It is my code:

var terminal=new Terminal({
            role: "server",
            shell: "linux",

            useStyle: true,
            convertEol: true,
            screenKeys: true,
            cursorBlink: true,
            visualBell: true,
        });
terminal.write(atob("w6k6IGNvbW1hbmQgbm90IGZvdW5k"));

Display: Ã© command not foun

From my terminal:
$ echo "w6k6IGNvbW1hbmQgbm90IGZvdW5k"|base64 -d|od -x
0000000 a9c3 203a 6f63 6d6d 6e61 2064 6f6e 2074
0000020 6f66 6e75 0064
0000025

Can you help me ?

Browser: Frefox linux 112.0 (64 bit)
OS ubuntu
Env: fr-FR.utf8

**Comment by jerch** [comment, marked as answer]
`atob` creates a "bytestring", where every byte maps to a character in [0..255], thus multibyte chars of the original UTF-8 string are not fused into higher codepoints yet.

`term.write` supports as input:
- UTF-8 byte sequences given as `Uint8Array`, or
- JS unicode string (interpreted as UTF-16)

(See https://xtermjs.org/docs/guides/encoding/ for encoding details.)

Now putting this together - you gonna need to convert the bytestring into an `Uint8Array` instance first, then either write that directly to the terminal or create a proper JS unicode string before writing:
```javascript
// convert to byte sequence
const sdata = atob("w6k6IGNvbW1hbmQgbm90IGZvdW5k");
const bytes = new Uint8Array(sdata.length);
for (let i = 0; i < bytes.length; ++i) {
  bytes[i] = sdata.charCodeAt(i);
}

// write bytes to terminal
term.write(bytes);

// write string to terminal
const td = new TextDecoder();
const s = td.decode(bytes);  // s is a normal JS string now
term.write(s);
```

**Comment by Archie1978** [comment]
ah, yes I have a little trouble with its binary in javascript.

Thanks you
